raft/raft.hh

```diff
@@ -56,61 +56,98 @@ using server_info = bytes;
 
 struct server_address {
     server_id id;
-    bool can_vote = true;
     server_info info;
+
+    server_address(server_id id, server_info info)
```

Comment:
Having this suggestion earlier would have been nice.
Let me see if using `explicit` here won't cause the code to stop compiling...

Reply:
Well, there are many places in the code where I would have to write out the constructor names, e.g. in tests:
```
cfg.enter_joint({{server_addr_from_id(id1), true}, {server_addr_from_id(id2), true}, {server_addr_from_id(id3), false}})
```
I don't think `explicit` is worth it.

Reply:
Isn't this the point of `explicit` after all ? :) The question is how dangerous is it to allow implicit conversion here.

Reply:
@kbr- we could add one more helper to these places perhaps?

Reply:
> The question is how dangerous is it to allow implicit conversion here.

That's the question. We could debate, or just move on. I propose let's move on.
The point of this PR is not to make the code perfect, but make it better than it was.
Can I leave the follow-ups with `explicit` or whatever to you?

Reply:
I do not insist on `explicit`, but I do notice that our code tend to prefer it. Here, since one of the types `server_info` is basically used within `server_address` only it may be less important.